### ISSUE TITLE
Improving the cheat sheet and general flow

### ISSUE THREAD

**Issue by BasileChretien** [pharmacologie-caen/vigicaen#79]
The following are comments I wrote when testing the cheat sheet.
I think that the package can do what it is expected to do. However its use might be hard for beginners, and I believe some points could be improved.
In my opinion, for improvement the most important task would be to reorganize a bit the functions so that the flow of use becomes evident for anyone.

My comments : 
In the actual version of the cheat sheet : 
- it is unclear that we have to convert the different tables to .parquet only once then store them 
--> Write that this step has to be done only once for each version of Vigibase ECL (updated 4 times per year on March, June, September and December), WHODrug and Meddra  (both updated twice a year on March and September) used.
--> has to be separated from the "load and assign tables" part
--> Create a Table conversion part : 
# Path ####
path_who <- "C:/Vigibase/VigiBase Extract Case Level 2024 Sep 1/who_ddx_sep_1_2024/"
path_meddra <- "C:/Meddra/MedDRA_27_1_English/Medascii/"
path_subs <- "C:/Vigibase/VigiBase Extract Case Level 2024 Sep 1/subsidiary_files_extract_sep_1_2024/"
path_main <- "C:/Vigibase/VigiBase Extract Case Level 2024 Sep 1/main_files_extract_sep_1_2024/"

## Table conversion ###
tb_main_sub(path_base = path_main,
                     path_sub = path_subs)
tb_who(path_who = path_who)
tb_meddra(path_meddra = path_meddra)

--> Example of tb_main could be improved (no need for the list, just explain you downloaded and unzipped Vigibase ecl under this format C:\Vigibase\VigiBase Extract Case Level 2024 Sep 1\main_files_extract_sep_1_2024 and set it as your path_main path. Same for the other paths
--> Do we really need a tb_sub if tb_main can do both?? It is a bit confusing. Maybe deprecate tb_sub and rename tb_main into tb_main_sub
--> Btw are you sure that .parquet is better than .fst in our context? I have read that for files weighting tens of GB .fst is faster than .parquet. I did not search much about it though.

For subsetting tables, need a quick explanation of the reason why you can/should subset tables.

For setting up the dictionary,  should precise that drugs should belong to the WHO drug tables, and that ADR should be chosen according to the Meddra tables (with the same level for all terms ; need to be done several times if several levels are chosen)

Could also precise that MedDRA_Id correspond to llt of the Meddra classification.

General flow : see issue #77.

Help of get_drecno : an f is missing in the following phrase : "and alitretinoin is not ound when looking for tretinoin."

Help of get_atc_code : for the example it is actually written atc_sel <-
     rlang::list2(l03_j01 = c("L03AA", "J01CA"),
                  c09aa = c("C09AA")
     )
     
     wouldn't id be better to write :
     atc_sel <-
     rlang::list2(penicillins_GCSF = c("L03AA", "J01CA"),
                  ACE_inhibitors = c("C09AA")
     )
     BTW I do not understand well the combination of GCSF with penicillins in our context as the result is an AND/OR and not an AND.
     
Help of check_dm : replace ex_$a_llt by d_list. Delete demo_
Same for help of compute_or_abcd

By the way, are we allowed to share the example data (with the _ at the end) ??? I am not sure that Uppsala gave the permission.

Precise that compute_or_mod is simply to format the results of a glm. Integrate the different steps included in the example directly into the function? And program it like compute_or_abcd with a z argument for adjustment factors?
     
 For desc_favar rather than value I would prefer to have several columns (one for n (with the percent option), and one for N). Also, it is unclear what the pad_width argument is for.

The creation of luda itself must be simplified a lot (automatized). I had trouble using desc_tto and extract_tto because at first I did not create it correctly.

For desc_tto we do not know what the data in "value" represent. Should be precised in the help (In general in all R help sections I like when I have a description of the output)

For extract_tto it works. But when I try to replicate the cheat sheet I obtain the following error : tto_2 <- extract_tto(.data = luda,
+                    adr_s = "heart disease",
+                    drug_s = "antalgics")
> 
> 
> hist(tto_2
+     )
Error in hist.default(tto_2) : 'x' must be numeric

--> btw, at first glance we have the feeling that extract_tto is designed to draw times to onset. Integrate hist directly into the function?

For the dechallenge and rechallenge function it would be very important to have the number of negative dechallenge as well, negative rechallenge as well.
Also for desc_rch, in the cheat sheet it is not written that a demo argument is needed.

For the desc_ family, in general I would prefer a unique and customizable function that allows to build a table 1 directly usable for scientific articles.
 
 Screen_adr could coutain an optional drug = argument

**Comment by cdolladille** [comment]
Good news, except for a few details, this issue is resolved. Most of exclusions/won't do are motivated in #73 
I'll move the remaining little things into separate issues
